pkg/runtime-manager/server/docker/server.go

```diff
@@ -0,0 +1,197 @@
+/*
+Copyright 2022 The Koordinator Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package docker
+
+import (
+	"bytes"
+	"context"
+	"net"
+	"net/http"
+	"net/http/httputil"
+	"net/url"
+	"regexp"
+
+	dockertypes "github.com/docker/docker/api/types"
+	"github.com/docker/docker/client"
+	"k8s.io/klog/v2"
+
+	"github.com/koordinator-sh/koordinator/apis/runtime/v1alpha1"
+	"github.com/koordinator-sh/koordinator/cmd/runtime-manager/options"
+	"github.com/koordinator-sh/koordinator/pkg/runtime-manager/dispatcher"
+	resource_executor "github.com/koordinator-sh/koordinator/pkg/runtime-manager/resource-executor"
+	types2 "github.com/koordinator-sh/koordinator/pkg/runtime-manager/server/types"
+	"github.com/koordinator-sh/koordinator/pkg/runtime-manager/store"
+)
+
+type RuntimeManagerDockerServer struct {
+	dispatcher   *dispatcher.RuntimeHookDispatcher
+	reverseProxy *httputil.ReverseProxy
+	router       map[*regexp.Regexp]func(context.Context, http.ResponseWriter, *http.Request)
+}
+
+func (d *RuntimeManagerDockerServer) Name() string {
+	return "RuntimeManagerDockerServer"
+}
+
+func NewRuntimeManagerDockerServer() *RuntimeManagerDockerServer {
+	criInterceptor := &RuntimeManagerDockerServer{
+		dispatcher: dispatcher.NewRuntimeDispatcher(),
+	}
+	criInterceptor.router = map[*regexp.Regexp]func(context.Context, http.ResponseWriter, *http.Request){
+		regexp.MustCompile("^/(v\\d\\.\\d+/)?containers(/\\w+)?/update$"): criInterceptor.HandleUpdateContainer,
```

Comment:
Can we replace this custom regular implementation with a more mature route package?